Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -281,7 +281,7 @@ class BrowserViewController: UIViewController {
         })
 
 
-        searchLoader = SearchLoader(history: profile.history, urlBar: urlBar)
+        searchLoader = SearchLoader(profile: profile, urlBar: urlBar)
```

Comment:
Yup. More importantly, it does so *on the thread that's dereferencing the property*, so we were doing the lazy init multiple times. My fix elsewhere for the lazy init stuff was to use `dispatch_once` to ensure safety.

All of the bugs fixed in this queue interact, so it's hard to point the finger at just this race versus the creation of multiple `SwiftData` instances versus…